Client/src/Pages/Infrastructure/index.jsx

```diff
@@ -10,21 +74,234 @@ import { useNavigate } from "react-router-dom";
  */
 
 function Infrastructure() {
-	const navigate = useNavigate();	
+	/* Adding this custom hook so we can avoid using the HOC approach that can lower performance (we are calling the admin logic N times on initializing the project. using a custom hook will cal it ass needed ) */
+	const isAdmin = useIsAdmin();
+	const theme = useTheme();
+
+	const navigate = useNavigate();
+	const navigateToCreate = () => navigate("/infrastructure/create");
+
+	const [page, setPage] = useState(0);
+	/* TODO refactor this, so it is not aware of the MUI implementation. First argument only exists because of MUI. This should require onlu the new page. Adapting for MUI should happen inside of table pagination component */
+	const handleChangePage = (_, newPage) => {
+		setPage(newPage);
+	};
+
+	const [rowsPerPage, setRowsPerPage] = useState(5);
+	const handleChangeRowsPerPage = (event) => {
+		setRowsPerPage(parseInt(event.target.value));
+		setPage(0);
+	};
+	const [monitorState, setMonitorState] = useState({ monitors: [], total: 0 });
+
+	const { authToken } = useSelector((state) => state.auth);
+	const user = jwtDecode(authToken);
+
+	const fetchMonitors = async () => {
+		try {
+			const response = await networkService.getMonitorsByTeamId({
+				authToken,
+				teamId: user.teamId,
+				limit: 1,
+				types: ["hardware"],
+				status: null,
+				checkOrder: "desc",
+				normalize: true,
+				page: page,
+				rowsPerPage: rowsPerPage,
+			});
+			console.log({ response });
+			setMonitorState({
+				monitors: response?.data?.data?.monitors ?? [],
+				total: response?.data?.data?.monitorCount ?? 0,
+			});
+		} catch (error) {
+			console.error(error);
+		}
+	};
+
+	useEffect(() => {
+		fetchMonitors();
+	}, [page, rowsPerPage]);
+
+	const { determineState } = useUtils();
+	const { monitors, total: totalMonitors } = monitorState;
+	const monitorsAsRows = monitors.map((monitor) => ({
```

Comment:
We're doubling up on the `map` here first mapping monitor to rows, then mapping rows to React elemetns further down.  I don't think there's any need to do this in a separate step which gives us `2n` operations instead of `n` :thinking: